scripts/pgpwave8_upgrade

```diff
@@ -0,0 +1,168 @@
+#!/bin/bash
+
+usage()
+{
+    cat <<EOF
+usage $0 -i IOCNAME [-d DEVICE] [-r] [-p FIRMWARE_PATH]
+
+Read or upgrade the firmware version of a pgpwave8 given its ioc.
+
+-i The ioc whose pgpwave8's firmware should be upgraded
+-d The path to the device, defaulting to /dev/datadev_0
+-r Print the firmware version only. Cannot be used with -p
+-p The path to the mcs file containing the new firmware.
+Cannot be used with -r. Some firmware images can be found
+in /cds/group/pcds/package/wave8/images.
+EOF
+}
+
+remote()
+{
+    #!/bin/bash
+    # shellcheck disable=SC1091
+    source /cds/group/pcds/pyps/conda/rogue/etc/profile.d/conda.sh
+    if ! conda activate rogue_"$ROGUE_VERSION"; then
+        echo "A rogue environment for this version is not available."
+        exit 1
+    fi
+    if [[ ! -e $DEVICE ]]; then
+        echo "The device $DEVICE does not exist."
+        exit 1
+    fi
+    export PATH=/cds/group/pcds/engineering_tools/latest-released/scripts/:$PATH
+    # shellcheck disable=SC2034
+    if ! status=$(/cds/group/pcds/engineering_tools/latest-released/scripts/imgr "$IOCNAME" --status); then
+        echo "Unable to get the status of $IOCNAME. Exiting."
+        exit 1
+    fi
+
+    # change this to a common path later
+    WAVE8_REPO=/cds/home/t/tjohnson/trunk/workarea/wave8
+    if [[ $READ_VERSION -eq 1 ]]; then
+        if [[ $status == "RUNNING" ]]; then
+            pv=$(/cds/group/pcds/engineering_tools/latest-released/scripts/ioctool "$IOCNAME" pvs | grep FpgaVersion_RBV | awk '{print substr($1,1,length($1)-1)}')
+            caget "$pv" -0x
+        else
+	    # shellcheck disable=SC2164
+	    cd "$(dirname "$0")"
+            python pgpwave8_version.py -l "$LANE" --dev "$DEVICE"
+        fi
+        exit 0
+    fi
+
+    if [[ $status == "RUNNING" ]] && ! /cds/group/pcds/engineering_tools/latest-released/scripts/imgr "$IOCNAME" --disable; then
+        echo "Unable to disable ioc $IOCNAME. Exiting."
+        exit 1
+    fi
+    if ! python $WAVE8_REPO/software/scripts/wave8LoadFpga.py --l "$LANE" --dev "$DEVICE" --mcs "$FWPATH"; then
+        echo "Firmware update failed."
+        exit 1
+    fi
+    if [[ $status == "RUNNING" ]] && ! /cds/group/pcds/engineering_tools/latest-released/scripts/imgr "$IOCNAME" --enable; then
+        echo "Unable to enable ioc $IOCNAME."
+        exit 1
+    fi
+    echo "CHECK THAT THE WAVE8 IOC IS WORKING."
+}
+
+# shellcheck disable=SC2034
+while getopts "rp:i:d:" option; do
+    case $option in
+        p)
+            FWPATH="$OPTARG"
+            ;;
+        i)
+            IOCNAME="$OPTARG"
+            ;;
+        d)
+            DEVICE="$OPTARG"
+            ;;
+        r)
+            READ_VERSION=1
+            ;;
+        ?)
+            usage
+            exit 1
+            ;;
+    esac
+done
+
+if [[ -z $IOCNAME ]]; then
+    echo "The ioc name must be provided with the -i flag."
+    usage
+    exit 1
+fi
+
+if [[ $READ_VERSION -ne 1 ]]; then
+    if [[ -z $FWPATH ]]; then
```

Comment:
I think we should have this default to the latest available firmware if nothing is provided, but notify the user that's what we're doing. This should be symlinked to "latest" in our firmware images directory.

Reply:
Sorry, I worded that poorly. What is was envisioning is something similar to how we do things for the EVR/TPR firmware, where we have a directory with all of our images, but also a symlink pointing to our latest image, e.g `/cds/group/pcds/package/wave8/images/latest`. This could be the default of this script.

Reply:
I added some images in /cds/group/pcds/package/wave8/images and symlinked latest to v2.8.0